common/testing/taskpoller/taskpoller.go

```diff
@@ -84,70 +96,139 @@ func New(
 	}
 }
 
-// PollWorkflowTask issues PollWorkflowTaskQueueRequests to obtain a new workflow task.
+// PollWorkflowTask creates a workflow task poller that uses the given PollWorkflowTaskQueueRequest.
 func (p *TaskPoller) PollWorkflowTask(
-	tv *testvars.TestVars,
-	funcs ...OptionFunc,
-) (*workflowservice.PollWorkflowTaskQueueResponse, error) {
-	p.t.Helper()
-	options := newOptions(tv, funcs)
-	ctx, cancel := newContext(options)
-	defer cancel()
-	return p.pollWorkflowTask(ctx, options)
+	req *workflowservice.PollWorkflowTaskQueueRequest,
+) *workflowTaskPoller {
+	return &workflowTaskPoller{TaskPoller: p, pollWorkflowTaskRequest: req}
```

Comment:
I don't understand this function.

Reply:
The name might need to change. Maybe `WithWorkflowTask` now that we've gotten rid of almost all options.

The idea is that it (a) creates a workflow-specific poller that (b) will use a specific poll request, provided by the caller (instead of the default). There's an example of what that looks like in the PR.

Reply:
We decided to leave things as-is and make improvement as we go.